src/psyclone/psyir/backend/fortran.py

```diff
@@ -410,12 +408,16 @@ def gen_indices(self, indices, var_name=None):
                 # Lower and upper bounds of an array declaration specified
                 # by literal constant, symbol reference, or computed dimension
                 lower_expression = self._visit(index.lower)
+                if isinstance(index.lower, IntrinsicCall):
+                    lower_expression = lower_expression[7:-1]
```

Comment:
Instead of this, I made the same solution as the initialisation value and add a dangling parent. This also prevents that the node is used anywhere else and change unexpectedly.

Reply:
Nice, thanks.